src/shared/bootspec.c

```diff
@@ -962,6 +963,23 @@ static int boot_entries_find_unified_addons(
         return 0;
 }
 
+static int boot_entries_find_unified_global_addons(
+                BootConfig *config,
+                const char *root,
+                const char *d_name) {
+
+        int r;
+        _cleanup_closedir_ DIR *d = NULL;
+
+        r = chase_and_opendir(root, NULL, CHASE_PROHIBIT_SYMLINKS, NULL, &d);
```

Comment:
Hmm, it is strange to create the `DIR` object and never use it. It should be possible to just use `chase_and_open` here and pass the fd directly.

Reply:
As far as I can tell `chase_and_opendir` does not return an `fd`. So I need the `DIR`.